src/SignalR/common/Http.Connections.Common/src/NegotiateProtocol.cs

```diff
@@ -13,77 +15,166 @@ namespace Microsoft.AspNetCore.Http.Connections
     public static class NegotiateProtocol
     {
         private const string ConnectionIdPropertyName = "connectionId";
+        private static readonly byte[] ConnectionIdPropertyNameBytes = Encoding.UTF8.GetBytes("connectionId");
```

Comment:
Consider re-using the const strings to avoid having to change them in multiple places.
```C#
private static readonly byte[] ConnectionIdPropertyNameBytes = Encoding.UTF8.GetBytes(ConnectionIdPropertyName);
```

Reply:
OR, if performance is really critical, pre-compute the UTF-8 bytes and store them as byte arrays whose values are known at compile-time.

For example, for "url":
```C#
private static ReadOnlySpan<byte> ConnectionIdPropertyNameBytes => new byte [3] {117, 114, 108};
// OR
private static ReadOnlySpan<byte> ConnectionIdPropertyNameBytes => new byte[3] { (byte)'u', (byte)'r', (byte)'l' };
```

Reply:
If that's something we decide is worthwhile, we would want to change a bunch of type in Kestrel like HttpProtocol to pre-compute bytes as well.